tezos/messages/src/p2p/encoding/operations_for_blocks.rs

```diff
@@ -25,7 +25,7 @@ use tezos_encoding::json_writer::JsonWriter;
 /// Serde serialization is still recursive and can lead to stack overflow.
 /// This value can be handled by serde and still it is big enough.
 /// to handle 2 ** 512 elements tree.
-pub const PATH_MAX_DEPTH: RecursiveDataSize = 512;
+pub const PATH_MAX_DEPTH: RecursiveDataSize = u16::MAX;
```

Comment:
Do we know if there is any limit (or if there is NO limit) in the OCaml decoders?

Reply:
As we discovered, the limit is 8 operations in tree, and consequently 3 links in path. Updated accordingly.